### PR TITLE
Colliding points mex

### PR DESCRIPTION
**Pull request by MarcoCar**
This PR adds the mex and m files needed to use RigidBodyManipulator::collidingPoints in Matlab. It also adds collidingPointsmex to the CMakeLists in ```drake/systems/plants```

### CODE REVIEW

**Comment by RussTedrake** [comment]
code looks good to me. thnx.
but we need some sort of unit test, so that the drake build servers have coverage on this.
(I'll also need to run the build servers again to make sure it does not conflict w/ the changes I just merged from @tkoolen)

**Comment by MarcoCar** [comment]
I also added another function that stops whenever a collision is found and return a boolean. This speeds up the computation in case we're only interested in a binary output. The test file checks the output of both functions using an atlas model and a sample point cloud.

File: drake/systems/plants/collision/BulletModel.cpp [hunk 1]
```diff
@@ -299,6 +299,40 @@ namespace DrakeCollision
...
 
+  bool BulletModel::collidingPointsCheckOnly(const vector<Vector3d>& points,
+                                             double collision_threshold)
+  {
+    // Create sphere geometry
+    btSphereShape bt_shape(collision_threshold);
+
+    // Create Bullet collision object
+    btCollisionObject bt_obj;
+    btCollisionObject* bt_obj_ptr = static_cast<btCollisionObject*>(&bt_obj);
```
Comment (avalenzu) [inline drake/systems/plants/collision/BulletModel.cpp]: Just out of curiosity, why is the cast necessary here? Shouldn't `&bt_obj` already be a `btCollisionObject*`?

Comment (MarcoCar) [inline drake/systems/plants/collision/BulletModel.cpp]: @avalenzu I just copied and changed the collidingPoints function to return a boolean, so I didn't notice that. I think you are the author of that function :) Should I investigate that further?

Comment (MarcoCar) [inline drake/systems/plants/collision/BulletModel.cpp]: @avalenzu yes, it doesn't seem to be necessary so I removed it. Thanks for reviewing the code.

Comment (avalenzu) [inline drake/systems/plants/collision/BulletModel.cpp]: Strange - why would I have done that? Anyway, thanks for fixing it!

**Comment by RussTedrake** [comment]
looks like perhaps you don't handle the case where bullet is not present?  (the windows build servers don't have bullet, so catch these for us)

**Comment by MarcoCar** [comment]
@RussTedrake I added the dependency checks. Can you run the tests again? thanks

**Comment by avalenzu** [comment]
If you close and reopen the pull request, that should trigger the build
servers

**Comment by RussTedrake** [comment]
Only for people on the white list.

**Comment by avalenzu** [comment]
Aha. Didn't know that.

**Comment by RussTedrake** [comment]
@MarcoCar -- please see the build errors?

**Comment by RussTedrake** [comment]
i've done the manual merge and will reopen on that branch.

**Pull request closed without merging**